src/public/modules/forms/admin/components/edit-logic.client.component.js

```diff
@@ -74,8 +81,19 @@ function editLogicComponentController($uibModal, FormFields) {
   }
 
   vm.deleteLogic = function (logicIndex) {
-    vm.myform.form_logics.splice(logicIndex, 1)
-    updateLogic({ form_logics: vm.myform.form_logics })
+    const logicIdToDelete = vm.myform.form_logics[logicIndex]._id
+    $q.when(AdminFormService.deleteFormLogic(vm.myform._id, logicIdToDelete))
+      .then((result) => {
+        if (result.request.status == 200) {
```

Comment:
I don't think this is necessary. `axios` will reject the Promise if a non-2xx status is returned anyway, so this chunk should only have to deal with the success case (i.e. you can go straight to `vm.myform.form_logics.splice(logicIndex, 1)`)

Reply:
thanks! edited